ddl/backfilling.go

```diff
@@ -583,9 +583,10 @@ func (dc *ddlCtx) sendTasksAndWait(scheduler *backfillScheduler, totalAddedCount
 		err = dc.isReorgRunnable(reorgInfo.Job.ID)
 	}
 
+	// Update the reorg handle that has been processed.
+	err1 := reorgInfo.UpdateReorgMeta(nextKey, scheduler.sessPool)
```

Comment:
I removed the `updateDDLReorgStartHandle` from `runReorgJob` on timeout and added it here instead.
It may change the number of transactions, but I think it is better to always save the state here, right after each batch is done (also still ignoring the error).